liquigraph-core/src/main/java/org/liquigraph/core/io/lock/LockableConnection.java

```diff
@@ -77,6 +94,7 @@ public static LockableConnection acquire(Connection delegate, LiquigraphLock loc
     @Override
     public void close() throws SQLException {
         lock.release(this);
+        delegate.setAutoCommit(previousAutoCommit);
```

Comment:
I've looked at the code of both versions of the JDBC driver, the same thing always happen: if the previous state was already `false`, we're not changing anything in the constructor or here; if the previous state was `true`, setting it back to `true` here will commit whatever was in the current transaction.

So I think it's an issue if you change autocommit without committing or rolling back first. I think the connection should be rolled back explicitly before resetting, just in case there are some pending queries. Not that it should happen, but still, better safe than sorry.